### PR TITLE
planner: prepared plan cache support limit

### PR DESCRIPTION
**Pull request by shichun-0415**
### What is changed, added or deleted? (Required)
Starting from v6.6, plan cache support supports limit.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12686
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by fzzf678 — APPROVED** [review]

**Comment by fzzf678** [comment]
LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: sql-prepared-plan-cache.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ In the current version of TiDB, if a `Prepare` statement meets any of the follow
 - The query contains the `ignore_plan_cache` hint, such as `select /*+ ignore_plan_cache() */ * from t`.
 - The query contains variables other than `?` (including system variables or user-defined variables), such as `select * from t where a>? and b>@x`.
 - The query contains the functions that cannot be cached: `database()`, `current_user`, `current_role`, `user`, `connection_id`, `last_insert_id`, `row_count`, `version`, and `like`.
-- The query contains `?` after `Limit`, such as `Limit ?` and `Limit 10, ?`. Such queries are not cached because the specific value of `?` has a great impact on query performance.
+- The query with `Count` being `?` after `Limit` and the `Count` value is greater than 10000.
```
Comment (Oreoxmt) [inline sql-prepared-plan-cache.md:26]:
```suggestion
- The query with a variable as the `LIMIT` parameter (`LIMIT ?`) and the variable value is greater than 10000.
```

**Review by qw4990 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 2]
```diff
@@ -1726,7 +1726,15 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
-- This variables specifies whether to use the pipeline execution algorithm for window functions.
+- This variable specifies whether to use the pipeline execution algorithm for window functions.
+
+### `tidb_enable_plan_cache_for_param_limit` <span class="version-mark">New in v6.6.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Type: Boolean
+- Default value: `ON`
+- This variable controls whether Prepared Plan Cache caches execution plans with `Count` being `?` after `Limit`. The default value is `ON`, which means Prepared Plan Cache supports caching such execution plans. Note that Prepared Plan Cache does not support caching execution plans with a `Count` that is greater than 10000.
```
Comment (shichun-0415) [inline system-variables.md:1737]:
```suggestion
- This variable controls whether Prepared Plan Cache caches execution plans with a variable as the `LIMIT` parameter (`LIMIT ?`). The default value is `ON`, which means Prepared Plan Cache supports caching such execution plans. Note that Prepared Plan Cache does not support caching execution plans with a variable that is greater than 10000.
```

Comment (shichun-0415) [reply system-variables.md:1737]: @qw4990 @Oreoxmt @fzzf678 PTAL

Comment (Oreoxmt) [reply system-variables.md:1737]: LGTM

**Pull request merged**